src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -0,0 +1,607 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/ric_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/op/util/binary_elementwise_arithmetic.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/log.hpp>
+#include "itt.hpp"
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ReverseInputChannelsFusion, "ReverseInputChannelsFusion", 0);
+
+namespace ngraph {
+namespace pass {
+namespace ric_attr {
+
+// Attribute describes RIC type which we propagate.
+// Also, it contains callback which can expand this attribute to the real RIC sub-graph.
+// In addition, attribute has some functionality and properties for propagation.
+class Attribute {
+public:
+    using callback_t = std::function<void(Input<Node>, const Attribute &)>;
+
+    Attribute(std::vector<int64_t> order, int64_t axis, bool is_final = false, bool is_initial = false)
+        : m_order(std::move(order)), m_axis(axis), m_is_final(is_final), m_is_initial(is_initial) {
+        m_can_be_fused.emplace_back(std::make_shared<bool>(true));
+    }
+
+    // Method which is used to create a copy of attribute for further propagation.
+    // TODO: can be removed and replaced with regular copy but we need to get rid of
+    //       is_initial flag and use some other way to detect original RIC output.
+    Attribute propagate() const {
+        Attribute attr(m_order, m_axis);
+        attr.m_can_be_fused = m_can_be_fused;
+        return attr;
+    }
+
+    bool operator==(const Attribute & other) const {
```

Comment:
is this needed?

Reply:
required by Any which is used inside rt info map